Question: How much does height difference between the fighters impact the fight outcome?

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('../../data/ufc-master.csv')

In [20]:
no_eq = (data['height_dif'] != 0) & (data['height_dif'] != -187.96)
#In addition to equal hights also removed an outlier that was spotted when checking values in height_dif column.
data_filtered = data[no_eq][['height_dif', 'Winner']]

How often does the taller fighter win?

In [18]:
taller_wins = 0
for i in range(len(data_filtered)):
    #Height difference is calculated as blue fighter - red righter
    if (data_filtered['height_dif'].iloc[i] > 0) & (data_filtered['Winner'].iloc[i] == 'Blue'):
        taller_wins += 1
    elif (data_filtered['height_dif'].iloc[i] < 0) & (data_filtered['Winner'].iloc[i] == 'Red'):
        taller_wins += 1
taller_win_ratio = taller_wins / len(data_filtered)
print("Taller fighter wins " + str(round(taller_win_ratio * 100, 2)) + "% of the time.")
print("Sample size (fights where the height of the fighters was different): " + str(len(data_filtered)))

Taller fighter wins 51.34% of the time.
Sample size (fights where the height of the fighters was different): 3699


As we can see taller fighter wins 51.3% of the time, which makes it seem like the height is not a significant factor, but next we are going to how and if the win percetange changes for different height differences.

In [35]:
data_filtered['height_dif'].value_counts().head(5)
results = []
sample_sizes = []
for i in range(1, 8, 1):
    if i == 1:
        height_filter = abs(data_filtered['height_dif']) <= 2.54
    elif i == 7:
        height_filter = abs(data_filtered['height_dif']) > 15.24
    else:
        height_filter = (abs(data_filtered['height_dif']) > (i - 1) * 2.54) & (abs(data_filtered['height_dif']) <= i * 2.54)
    data_height = data_filtered[height_filter]
    sample_sizes.append(len(data_height))
    taller_wins_2 = 0
    for j in range(len(data_height)):
        if (data_height['height_dif'].iloc[j] > 0) & (data_height['Winner'].iloc[j] == 'Blue'):
            taller_wins_2 += 1
        elif (data_height['height_dif'].iloc[j] < 0) & (data_height['Winner'].iloc[j] == 'Red'):
            taller_wins_2 += 1
    results.append(taller_wins_2 / len(data_height))

In [36]:
for i in range(len(results)):
    if i == 0:
        print("When the height difference is between (0, 2.54] the taller fighter wins " + 
              str(round(results[i] * 100, 2)) + "% of the time.")
        print("Sample size: " + str(sample_sizes[i]) + "\n")
    elif i == len(results) - 1:
        print("When the height difference between the fighters is greater than 15.24 cm the taller fighter wins " +
             str(round(results[i] * 100, 2)) + "% of the time.")
        print("Sample size: " + str(sample_sizes[i]) + "\n")
    else:
        print("When the height difference between the fighters is between (" + str(i * 2.54) + ", " +
              str((i + 1) * 2.54) + "] the taller fighter wins " + str(round(results[i] * 100, 2)) + "% of the time")
        print("Sample size: " + str(sample_sizes[i]) + "\n")

When the height difference is between (0, 2.54] the taller fighter wins 49.89% of the time.
Sample size: 1327

When the height difference between the fighters is between (2.54, 5.08] the taller fighter wins 51.25% of the time
Sample size: 1001

When the height difference between the fighters is between (5.08, 7.62] the taller fighter wins 50.98% of the time
Sample size: 661

When the height difference between the fighters is between (7.62, 10.16] the taller fighter wins 57.38% of the time
Sample size: 366

When the height difference between the fighters is between (10.16, 12.7] the taller fighter wins 50.98% of the time
Sample size: 204

When the height difference between the fighters is between (12.7, 15.24] the taller fighter wins 49.41% of the time
Sample size: 85

When the height difference between the fighters is greater than 15.24 cm the taller fighter wins 56.36% of the time.
Sample size: 55



From these results we can see that being a taller fighter improves the chances of winning slighlty, but the benefit is certainly not comparable to the reach difference, where the results were quite one sided.